client/src/components/pages/RespondentQuiz.js

```diff
@@ -0,0 +1,113 @@
+import React, { useEffect, useState, useCallback } from 'react';
+import { useParams } from 'react-router-dom';
+import { Container, Row, Col, Form, Button } from 'react-bootstrap';
+
+import { useRespondent, getRespondentQuiz, takeQuiz } from '../../context/respondent/RespondentState';
+
+const RespondentQuiz = () => {
+    const [respondentState, respondentDispatch] = useRespondent();
+    const { error, respondent, quiz } = respondentState;
+    const { userId, quizId } = useParams();
+
+    const [questionsAnswered, setQuestionsAnswered] = useState([{
+        question_id: '',
+        answerGiven: ''
+    }]);
+
+    // let time = quiz && quiz.timeLimit;
+
+    // const [stateTimeLimit, setStateTimeLimit] = useState(time);
+
+    // const onTimeLimitChange = useCallback(() => {
+    //     setInterval(() => setStateTimeLimit(stateTimeLimit - 1), 10000);
+    // },[stateTimeLimit],
+    // );
+
+    // const clearTimer = () => {
+    //     clearInterval(stateTimeLimit);
+    // }
+
+    useEffect(() => {
+        if(error) {
+            console.log(error);
+        }
+        // onTimeLimitChange();
+        getRespondentQuiz(respondentDispatch, userId, quizId);
+    }, [respondentDispatch, error, userId, quizId]);
+
+    const onChange = (e, q, i) => {
+        if(e.target.checked) {
+            if(!questionsAnswered[i]['answerGiven'].includes(e.target.value)) {
+                setQuestionsAnswered([...questionsAnswered, 
+                    { ...questionsAnswered[questionsAnswered.length], 
+                        question_id: q._id, 
```

Comment:
addressing your concern on teams, i was wondering if something like questionsAnswer[i]['answerGiven'] = e.target.value on change would work, maybe you already tried. Here's also something I found https://stackoverflow.com/questions/27784212/how-to-use-radio-buttons-in-reactjs